src/python/pants/backend/docker/util_rules/dependencies.py

```diff
@@ -23,23 +17,19 @@ class InjectDockerDependencies(InjectDependenciesRequest):
 @rule
 async def inject_docker_dependencies(request: InjectDockerDependencies) -> InjectedDependencies:
     """Inspects COPY instructions in the Dockerfile for references to known targets."""
-    original_tgt = await Get(WrappedTarget, Address, request.dependencies_field.address)
-    sources = original_tgt.target[DockerImageSourceField]
-    if not sources.value:
-        return InjectedDependencies()
-
-    dockerfile = await Get(DockerfileInfo, DockerImageSourceField, sources)
+    dockerfile_info = await Get(
+        DockerfileInfo, DockerfileInfoRequest(request.dependencies_field.address)
+    )
     targets = await Get(
         Targets,
         UnparsedAddressInputs(
-            dockerfile.putative_target_addresses,
+            dockerfile_info.putative_target_addresses,
             owning_address=None,
         ),
     )
 
-    return InjectedDependencies(
-        Addresses([tgt.address for tgt in targets if PexBinaryFieldSet.is_applicable(tgt)])
-    )
+    referenced_targets = [tgt.address for tgt in targets if PexBinaryFieldSet.is_applicable(tgt)]
```

Comment:
Better to use a generator expression

```suggestion
    referenced_targets = (tgt.address for tgt in targets if PexBinaryFieldSet.is_applicable(tgt))
```

Reply:
True. This is a remnant from earlier version, where there was a conditional block before returning, that could append additional targets to this list.